### PR TITLE
Add missing functions for PixbufAnimation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @sdroege @EPashkin

### CODE REVIEW

File: src/animation.rs [hunk 1]
```diff
@@ -92,6 +95,93 @@ impl PixbufAnimation {
...
+    /// use gdk_pixbuf::PixbufAnimation;
+    ///
+    /// let f = File::open("some_file").expect("failed to open animation");
+    /// let pixbuf = PixbufAnimation::from_read(f).expect("failed to load animation");
+    /// ```
+    pub fn from_read<R: Read + Send + 'static>(r: R) -> Result<PixbufAnimation, Error> {
+        PixbufAnimation::from_stream(&gio::ReadInputStream::new(r), None::<&gio::Cancellable>)
+    }
+
+    pub fn from_stream<P: IsA<gio::InputStream>, Q: IsA<gio::Cancellable>>(
```
Comment (sdroege) [inline src/animation.rs:113]: Why can't this be autogenerated?

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise, but this should be autogenerated really

**Comment by sdroege** [comment]
@GuillaumeGomez Ping? :)

File: src/pixbuf_animation.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use PixbufAnimation;
+use PixbufAnimationIter;
+use glib::object::IsA;
+use glib::translate::*;
+use glib::TimeVal;
+
+pub trait PixbufAnimationExtManual {
+    fn get_iter(&self, start_time: Option<TimeVal>) -> PixbufAnimationIter;
```
Comment (sdroege) [inline src/pixbuf_animation.rs:12]:
```suggestion
    fn get_iter(&self, start_time: Option<&TimeVal>) -> PixbufAnimationIter;
```

Probably also for the others. But maybe there's a better Rust type for these things?

In `std` (non-public) there's:

```rust
    impl From<libc::timeval> for SystemTime {
        fn from(t: libc::timeval) -> SystemTime {
            SystemTime::from(libc::timespec {
                tv_sec: t.tv_sec,
                tv_nsec: (t.tv_usec * 1000) as libc::c_long,
            })
        }
    }
```

So maybe `SystemTime` would be a good choice here

File: src/pixbuf_animation_iter.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+glib_wrapper! {
+    pub struct PixbufAnimationIter(Object<gdk_pixbuf_sys::GdkPixbufAnimationIter, PixbufAnimationIterClass>);
+
+    match fn {
+        get_type => || gdk_pixbuf_sys::gdk_pixbuf_animation_iter_get_type(),
+    }
+}
+
+impl PixbufAnimationIter {
+    pub fn advance(&self, start_time: TimeVal) -> bool {
```
Comment (sdroege) [inline src/pixbuf_animation_iter.rs:19]:
```suggestion
    pub fn advance(&self, start_time: &TimeVal) -> bool {
```

Comment (GuillaumeGomez) [reply src/pixbuf_animation_iter.rs:19]: `TimeVal` is `Copy`, no?

File: src/pixbuf_animation.rs (hunk 2 above)
Comment (sdroege) [reply src/pixbuf_animation.rs:12]: See also https://github.com/gtk-rs/gio/issues/320

File: src/pixbuf_animation_iter.rs (hunk 3 above)
Comment (sdroege) [reply src/pixbuf_animation_iter.rs:19]: Yes

Comment (GuillaumeGomez) [reply src/pixbuf_animation_iter.rs:19]: Do you still suggest we should take a reference then? XD

Comment (sdroege) [reply src/pixbuf_animation_iter.rs:19]: No :)

**Comment by GuillaumeGomez** [comment]
Updated!

**Review by sdroege — APPROVED** [review]

**Pull request merged**